src/package/subscriberPackageVersion.ts

```diff
@@ -0,0 +1,503 @@
+    this.Profiles = await this.getFieldValue<SPV['Profiles']>('Profiles');
+    return this.Profiles;
+  }
+
+  public async getDependencies(): Promise<PackagingSObjects.SubscriberPackageDependencies> {
+    this.Dependencies = await this.getFieldValue<SPV['Dependencies']>('Dependencies');
+    return this.Dependencies;
+  }
+
+  public async getInstallValidationStatus(): Promise<PackagingSObjects.InstallValidationStatus> {
+    this.InstallValidationStatus = await this.getFieldValue<SPV['InstallValidationStatus']>('InstallValidationStatus');
+    return this.InstallValidationStatus;
+  }
+
+  /**
+   * Get the SubscriberPackageVersion SObject data for this SubscriberPackageVersion.
+   *
+   * @param force - force a refresh of the subscriber package version data.
+   * @returns {PackagingSObjects.SubscriberPackageVersion} SObject data.
+   */
+  public async getData(
+    options: { force?: boolean; includeHighCostFields?: boolean } = { force: false, includeHighCostFields: false }
+  ): Promise<PackagingSObjects.SubscriberPackageVersion> {
+    if (!this.data || options.force || options.includeHighCostFields) {
+      const queryFields = this.getFieldsForQuery(options);
+      if (queryFields.length === 0) {
+        return this.data;
+      }
+      try {
+        const queryNoKey = `SELECT ${queryFields.toString()} FROM SubscriberPackageVersion WHERE Id ='${await this.getId()}'`;
+        const escapedInstallationKey = this.password ? escapeInstallationKey(this.password) : null;
+        const queryWithKey = `${queryNoKey} AND InstallationKey ='${escapedInstallationKey}'`;
+        this.data = await this.connection.singleRecordQuery<PackagingSObjects.SubscriberPackageVersion>(queryWithKey, {
+          tooling: true,
+        });
+      } catch (err) {
+        throw messages.createError('errorInvalidIdNoRecordFound', [this.options.id], undefined, err as Error);
+      }
+
+      // map the fields returned from the query to the class properties
+      this.mapFields(queryFields);
+    }
+    return this.data;
+  }
+
+  /**
+   * Installs a package version in a subscriber org.
+   *
+   * Package Version install emits the following events:
+   * - PackageEvents.install.warning
+   * - PackageEvents.install.presend
+   * - PackageEvents.install.postsend
+   * - PackageEvents.install['subscriber-status']
+   *
+   * @param pkgInstallCreateRequest
+   * @param options
+   */
+  public async install(
+    pkgInstallCreateRequest: PackageInstallCreateRequest,
+    options: PackageInstallOptions = allZeroesInstallOptions
+  ): Promise<PackagingSObjects.PackageInstallRequest> {
+    try {
+      // before starting the install, check to see if the package version is available for install
+      await waitForPublish(
+        this.connection,
+        await this.getId(),
+        options.publishFrequency,
+        options.publishTimeout,
+        pkgInstallCreateRequest.Password
+      );
```

Comment:
Shouldn't `waitForPublish` only be run if an option is set?  Otherwise, the command parameter in the plugin is meaningless.

Reply:
@shetzel the check to skip the wait is in the waitForPublish function, same with installStatus